engine/api/worker/model_import.go

```diff
@@ -55,41 +51,33 @@ currentUGroup:
 		return nil, sdk.ErrWorkerModelNoAdmin
 	}
 
+	var badRequestError error
+	asSimpleUser := !u.Admin && !sdkWm.Restricted
```

Comment:
The following code is not really clear, its hard to understand what are the rules for a admin vs normal user and this is mixed with restricted and pattern usage...

Reply:
ok